src/middleware/axum.rs

```diff
@@ -27,6 +43,11 @@ pub struct OtelAxumLayer<P> {
 
 // add a builder like api
 impl<P> OtelAxumLayer<P> {
```

Comment:
Maybe we should mention somehow that `OtelAxumLayer` is generic over `MatchedPath` struct from an axum framework used by the service.

This way `OtelAxumLayer` could be generic and not tied to any specific version of `axum`.